lang/src/OTS/IMP/org/partiql/ots/legacy/types/TimeType.kt

```diff
@@ -1,14 +1,37 @@
 package OTS.IMP.org.partiql.ots.legacy.types
 
-import OTS.ITF.org.partiql.ots.TypeParameters
 import OTS.ITF.org.partiql.ots.type.ScalarType
+import OTS.ITF.org.partiql.ots.type.TypeParameters
 import org.partiql.lang.eval.ExprValueType
 
+/**
+ * We might need to remove [withTimeZone] in TIME type and separate it into TIME_WITH_TIME_ZONE
+ * type & TIME_WITHOUT_TIME_ZONE type later
+ *
+ * The current modeling inherits the implementation of TIME type in the `main` branch
+ */
 data class TimeType(
     val withTimeZone: Boolean = false
 ) : ScalarType {
     override val id: String
-        get() = "time"
+        get() = when (withTimeZone) {
+            true -> "time_with_time_zone"
+            false -> "time"
+        }
+
+    override val names: List<String>
+        get() = when (withTimeZone) {
+            true -> listOf("time with time zone")
+            false -> listOf("time")
+        }
+
+    override fun validateParameters(typeParameters: TypeParameters) {
+        when (typeParameters.size) {
+            0 -> {}
+            1 -> require(typeParameters[0] > 0)
+            2 -> error("$id type requires at most 1 parameter")
+        }
+    }
```

Comment:
Opinion, but I think it would be cleaner to just have two different classes.

TimeType and TimeWithTimeZoneType or something like that.

Reply:
I think this is a good idea, but this is a refactoring to `Time` type. Do you think we can setup another PR for it, if needed?